testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/FilesystemRealmEncryptedTestCase.java

```diff
@@ -0,0 +1,200 @@
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.api.ServerSetup;
+import org.jboss.as.arquillian.api.ServerSetupTask;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.test.integration.management.util.CLIWrapper;
+import org.jboss.as.test.integration.security.common.Utils;
+import org.jboss.as.test.integration.security.common.servlets.SimpleSecuredServlet;
+import org.jboss.as.test.integration.security.common.servlets.SimpleServlet;
+import org.jboss.as.test.shared.ServerReload;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Full authentication tests for Elytron Encrypted Filesystem Realm.
+ *
+ * @author Ashpan Raskar <araskar@redhat.com>
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup({FilesystemRealmEncryptedTestCase.SetUpTask.class})
+public class FilesystemRealmEncryptedTestCase {
+
+    private static final String DEPLOYMENT_ENCRYPTED = "filesystemRealmEncrypted";
+    private static final String USER = "plainUser";
+    private static final String PASSWORD = "secretPassword";
+
+
+    @Deployment(name = DEPLOYMENT_ENCRYPTED)
+    public static WebArchive deploymentWithCharsetAndEncoded() {
+        final WebArchive war = ShrinkWrap.create(WebArchive.class, DEPLOYMENT_ENCRYPTED + ".war");
+        war.addClasses(SimpleServlet.class);
+        war.addClasses(SimpleSecuredServlet.class);
+        war.addAsWebInfResource(FilesystemRealmEncryptedTestCase.class.getPackage(), "filesystem-realm-web.xml", "web.xml");
+        war.addAsWebInfResource(Utils.getJBossWebXmlAsset(DEPLOYMENT_ENCRYPTED), "jboss-web.xml");
+        return war;
+    }
+
+    /**
+     *
+     * Test Filesystem realm when choosing to encrypt it.
+     */
+    @Test
+    @OperateOnDeployment(DEPLOYMENT_ENCRYPTED)
+    public void testEncryptionPass(@ArquillianResource URL webAppURL) throws Exception {
+        URL url = prepareURL(webAppURL);
+        Utils.makeCallWithBasicAuthn(url, USER, PASSWORD, SC_OK);
+    }
+
+    /**
+     *
+     * Test Filesystem realm when choosing to encrypt it with a bad credential.
+     */
+    @Test
+    @OperateOnDeployment(DEPLOYMENT_ENCRYPTED)
+    public void testEncryptionCredentialFail(@ArquillianResource URL webAppURL) throws Exception {
+        URL url = prepareURL(webAppURL);
+        Utils.makeCallWithBasicAuthn(url, USER+"123", PASSWORD, SC_UNAUTHORIZED);
+    }
+
+    /**
+     *
+     * Test Filesystem realm when choosing to encrypt it with a bad Secret Key.
+     */
+    @Test
+    @OperateOnDeployment(DEPLOYMENT_ENCRYPTED)
+    public void testEncryptionSecretKeyFail(@ArquillianResource URL webAppURL) throws Exception {
+        URL url = prepareURL(webAppURL);
+        SetUpTask.setUpBadCredentialStore(SetUpTask.REALM_NAME, SetUpTask.CREDENTIAL_STORE+"New");
```

Comment:
I don't understand why this test passes sometimes - there is no `mycredstoreNew` resource, hence this test should fail here because the call (writing the attribute) should fail.

Reply:
I changed the implementation of creating the secondary Credential Store resource to be in the `setUpTestDomain(...)` method. This way the resource is always present

https://github.com/ashpan/wildfly/blob/WFLY-15075-encrypt/testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/FilesystemRealmEncryptedTestCase.java#L148

Reply:
I'm sorry, I see the resource is set up at the last line of `setUpTestDomain`. That means that the whole test case fails sometimes, going deeper...

Reply:
I suspect [LdapRealmTestCase](https://github.com/wildfly/wildfly/blob/3eaeac0b07b4a03917f94bfb8368b6ee09c12530/testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/LdapRealmTestCase.java#L340) of setting the `from-roles-attribute` without proper cleaning. I don't have time to verify this today, I can continue tomorrow if no one will take this over.

Reply:
Removed `from-roles-attribute` resource during teardown of `LdapRealmTestCase`

https://github.com/Ashpan/wildfly/blob/6c25eb8dc9861961713ad7ce934b9864f8e4f1b2/testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/LdapRealmTestCase.java#L363

Reply:
Thank you. But it seem it was a wrong guess, this test case is still failing on some combinations.

Reply:
IIUIC, `LdapRealmTestCase` and `FilesystemRealmTestCase` are also affected, but there are no related negative tests, hence the setup failure is not visible and tests pass with unsecured configuration.

Reply:
Maybe we can workaround this by renaming the `from-roles-attribute` resource, and create a new Jira issue for the other test cases.